### ISSUE TITLE
don't same attack performance result BFA paper(resnet44 and resnet 56)

### ISSUE THREAD

**Issue by pdh930105** [elliothe/BFA#5]
I show that attack performance differs from BFA papers while attempting resnet44 and resnet56 BFA Attack. (paper resnet44 8bit is [6, 10, 11, 13, 22] but my resnet44 8bit is [57, 56, 59, 62, 60] and not drop acc 10%)

Therefore, I think BFA attack performance doesn't come out properly when the network gets deeper, what do you think?

Also, if possible, can you provide the weight parameter you used?

my resnet44 and resnet56 training script is this is this.
`
model=resnet56_quan
dataset=cifar10
epochs=160
train_batch_size=128
test_batch_size=128
optimizer=SGD
quan_bitwidth=8
seed=1
label_info=train

save_path=./save/${DATE}/${dataset}_${model}_${epochs}_${optimizer}_${label_info}_${quan_bitwidth}_${seed}
tb_path=${save_path}/tb_log  #tensorboard log path

############### Neural network ############################
{
/usr/bin/env python ./main.py --dataset ${dataset} \
    --data_path ${data_path}   \
    --arch ${model} --save_path ${save_path} \
    --quan_bitwidth ${quan_bitwidth} \
    --manualSeed ${seed} \
    --epochs ${epochs} --learning_rate 0.1 \
    --optimizer ${optimizer} \
        --schedule 80 120  --gammas 0.1 0.1 \
    --attack_sample_size ${train_batch_size} \
    --test_batch_size ${test_batch_size} \
    --workers 4 --ngpu 1 --gpu_id 0 \
    --print_freq 100 --decay 0.0003 --momentum 0.9
} &

model=resnet44_quan
dataset=cifar10
epochs=160
train_batch_size=128
test_batch_size=128
optimizer=SGD
quan_bitwidth=8
seed=1
label_info=train

save_path=./save/${DATE}/${dataset}_${model}_${epochs}_${optimizer}_${label_info}_${quan_bitwidth}_${seed}
tb_path=${save_path}/tb_log  #tensorboard log path

############### Neural network ############################
{
/usr/bin/env python ./main.py --dataset ${dataset} \
    --data_path ${data_path}   \
    --arch ${model} --save_path ${save_path} \
    --quan_bitwidth ${quan_bitwidth} \
    --manualSeed ${seed} \
    --epochs ${epochs} --learning_rate 0.1 \
    --optimizer ${optimizer} \
        --schedule 80 120  --gammas 0.1 0.1 \
    --attack_sample_size ${train_batch_size} \
    --test_batch_size ${test_batch_size} \
    --workers 4 --ngpu 1 --gpu_id 0 \
    --print_freq 100 --decay 0.0003 --momentum 0.9
} &

**Comment by pdh930105** [comment]
sorry my mistake.

https://github.com/elliothe/BFA/blob/6ad210cbf12c2db0a1c1d89cb210b39ef5cf62fa/utils.py#L31

change_quan_bitwidth(m, 8) is return unsigned integer type m.b_w([[128], [64], [32], [16], [8], [4], [2], [1]]) so I modified this code, what do you thinks?

original
m.b_w[0] = -m.b_w[0]
modified
m.b_w[0] = m.b_w[0] * torch.sign(m.b_w[0]) * -1 (MSB is always minus position)
